src/System.Management.Automation/engine/remoting/client/remotepipeline.cs

```diff
@@ -912,8 +912,7 @@ private void InitPowerShellForConnect(bool syncCall)
 
             // Check to see if this pipeline already exists in the runspace.
             RemotePipeline currentPipeline = (RemotePipeline)((RemoteRunspace)_runspace).GetCurrentlyRunningPipeline();
-            if (currentPipeline == null ||
-                currentPipeline != null && !ReferenceEquals(currentPipeline, this))
+            if (currentPipeline == null || !ReferenceEquals(currentPipeline, this))
```

Comment:
I think `currentPipeline == null` is redundant with `ReferenceEquals` here.

Reply:
Seems like.